pkg/controller/backup_controller.go

```diff
@@ -80,30 +79,28 @@ func NewBackupController(
 	backupInformer informers.BackupInformer,
 	client arkv1client.BackupsGetter,
 	backupper backup.Backupper,
-	objectStoreConfig api.CloudProviderConfig,
-	bucket string,
 	pvProviderExists bool,
 	logger logrus.FieldLogger,
 	logLevel logrus.Level,
 	pluginRegistry plugin.Registry,
 	backupTracker BackupTracker,
+	backupLocationInformer informers.BackupStorageLocationInformer,
 	metrics *metrics.ServerMetrics,
 ) Interface {
 	c := &backupController{
-		backupper:         backupper,
-		objectStoreConfig: objectStoreConfig,
-		bucket:            bucket,
-		pvProviderExists:  pvProviderExists,
-		lister:            backupInformer.Lister(),
-		listerSynced:      backupInformer.Informer().HasSynced,
-		client:            client,
-		queue:             workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "backup"),
-		clock:             &clock.RealClock{},
-		logger:            logger,
-		logLevel:          logLevel,
-		pluginRegistry:    pluginRegistry,
-		backupTracker:     backupTracker,
-		metrics:           metrics,
+		backupper:            backupper,
+		pvProviderExists:     pvProviderExists,
+		lister:               backupInformer.Lister(),
+		listerSynced:         backupInformer.Informer().HasSynced,
+		client:               client,
+		queue:                workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "backup"),
+		clock:                &clock.RealClock{},
+		logger:               logger,
+		logLevel:             logLevel,
+		pluginRegistry:       pluginRegistry,
+		backupTracker:        backupTracker,
+		backupLocationLister: backupLocationInformer.Lister(),
```

Comment:
you'll need to wait for the `backupLocationInformer` to sync in the `Run()` func like is being done for the `backupInformer`

Reply:
Is this just using a `Synced` field and updating the condition on L165 to look for it? Or do I need a separate condition?

Reply:
yep add a `backupLocationListerSynced` field to the struct, set it in the constructor, then add it as an arg to the `WaitForCacheSync(...)` call (it's variadic)